x-pack/legacy/plugins/apm/public/components/shared/charts/CustomPlot/Legends.js

```diff
@@ -75,7 +75,7 @@ export default function Legends({
   seriesEnabledState,
   truncateLegends
 }) {
-  if (noHits) {
+  if (noHits && seriesEnabledState.filter(Boolean).length === 0) {
```

Comment:
I think it's the `noHits` that needs fixin'. We also pass `noHits` to `StaticPlot` which currently also has a minor glitch (it shows "No data within this time range") and the y-axis is removed:

<img width="560" alt="Screen Shot 2019-08-21 at 21 15 51" src="https://user-images.githubusercontent.com/209966/63461452-f1898b00-c458-11e9-8b35-79109ce74b28.png">

I think it's better like this:
<img width="548" alt="Screen Shot 2019-08-21 at 21 16 01" src="https://user-images.githubusercontent.com/209966/63461485-0403c480-c459-11e9-9692-ef8413892330.png">

(yes, the blank screen look a little odd here, but the transition when toggling legends is nicer because the y-axis doesn't disappear and the empty state text doesn't appear).

Reply:
I think we decided to drop this because it was just so hard to get right: https://github.com/elastic/kibana/pull/40353#discussion_r301242860.

Do you mean that displaying the no data text is a glitch? Can you elaborate?

Reply:
I think the problem is that the `noHits` check is using the `enabledSeries` (enabled in the legends) where it should use the raw `series` instead.

Changing this:
https://github.com/elastic/kibana/blob/efb27e4ebda11255a730cfb7767cfd0ba63e544c/x-pack/legacy/plugins/apm/public/components/shared/charts/CustomPlot/index.js#L141-L144

To this:
```js
    const hasValidCoordinates = flatten(series.map(s => s.data)).some(p =>
      isValidCoordinateValue(p.y)
    );
    const noHits = !hasValidCoordinates;
```

should fix it afaict

Reply:
Yeah, looks like it works! I've tried various things to break it and everything seems fine. Makes me wonder why we struggled previously to get it to work properly 🤔